modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -5,6 +5,151 @@
+    // the InstantArticle object.
+    $context = array(
+      'entity_type' => $type,
+      'entity' => $entity,
+    );
+    $entity->fb_instant_article = \Drupal\fb_instant_articles\ArticleWrapper::create($context)->getArticle();
+  }
+}
+
+/**
+ * Implements hook_fb_instant_articles_article_alter().
+ *
+ * @see fb_instant_articles_display_module_implements_alter()
+ */
+function fb_instant_articles_display_fb_instant_articles_article_alter($instantArticle, $context) {
+  if (!isset($context['entity_type']) || !isset($context['entity'])) {
+    return FALSE;
+  }
+
+  // We map the Entity properties here, so we can ensure it runs only once per
+  // Entity (during load). If other modules wish to alter this, they should
+  // implement hook_fb_instant_articles_article_alter().
+  \Drupal\fb_instant_articles_display\EntityPropertyMapper::create($context['entity_type'], $context['entity'], $instantArticle)
+    ->map();
+}
+
+/**
+ * Implements hook_module_implements_alter().
+ *
+ * @see fb_instant_articles_display_declare_entity_preprocess_hooks()
+ */
+function fb_instant_articles_display_module_implements_alter(&$implementations, $hook) {
+  // Dynamically declare hook_preprocess_ENTITY() for each Entity type. This may
+  // be as good a place as any to call this function.
+  // @see module_implements()
+  fb_instant_articles_display_declare_entity_preprocess_hooks();
+
+  // We also ensure hook_fb_instant_articles_article_alter() runs first, so that
+  // other modules can override mappings made by this module by implementing
+  // hook_fb_instant_articles_article_alter().
+  if ($hook != 'fb_instant_articles_article_alter') {
+    return;
+  }
+  $module = 'fb_instant_articles_display';
+  $group = array($module => $implementations[$module]);
+  unset($implementations[$module]);
+  $implementations = $group + $implementations;
+}
+
+/**
+ * Dynamically declare hook_preprocess_ENTITY() for each Entity type.
+ *
+ * This adds a theme function to control the output of entity_view(), for every
+ * entity type, when the view mode is 'fb_instant_article'. See
+ * ctools_features_declare_functions() for precedent on dynamically declared
+ * functions.
+ *
+ * See @link https://www.drupal.org/node/1443202#comment-5787770 this d.o issue comment about updating documention for hook_preprocess_ENTITY() in D8. @endlink
+ *
+ * @see hook_preprocess_HOOK()
+ * @see fb_instant_articles_display_module_implements_alter()
+ * @see fb_instant_articles_display_theme()
+ * @see theme_entity__fb_instant_article()
+ */
+function fb_instant_articles_display_declare_entity_preprocess_hooks() {
+  foreach (array_keys(entity_get_info()) as $entity_type) {
+    $code = '';
+    if (!function_exists("fb_instant_articles_display_preprocess_{$entity_type}")) {
+      $code .= 'function fb_instant_articles_display_preprocess_' . $entity_type . "(&" . '$vars' . ") {
+      if (" . '$vars' . "['view_mode'] === 'fb_instant_article') {
```

Comment:
For some entites `$vars['view_mode']` is undefined so we get PHP notices.  Needs an `isset` call.